kernel/thread/thread.c

```diff
@@ -851,10 +851,9 @@ int thd_join(kthread_t *thd, void **value_ptr) {
     if(thd == thd_current)
         return -4;
 
-    if((rv = irq_inside_int())) {
+    if(irq_inside_int()) {
         dbglog(DBG_WARNING, "thd_join(%p) called inside an interrupt with "
-               "code: %x evt: %.4x\n", (void *)thd, ((rv >> 16) & 0xf),
-               (rv & 0xffff));
+               "code: [%x]\n", (void *)thd, irq_active_int(0));
         return -1;
```

Comment:
> I was wrong on this (I think). Confused level and depth, which are the reverse of each other.
Ah yes, you are correct here. I've used the terms `depth` vs `level` for `irq_int_depth()` and `irq_active_int()` to draw that distinction.

I was also unsure about potentially losing the `evt` ID for the code, but... The truth is all you need is the code to fully identify an interrupt, because, assuming you need it, you can always look up the `evt` ID for it... Just wasn't sure whether it warranted an entirely new API with a new `evt_t` (to be similarly proper) to add such information, but... if you want, I can do it.

Reply:
OH. I did think that actually making that detail public would further add to the confusion of distinguishing between "general exceptions," and "interrupts," because then we're mixing the API which has opted for IRQ to be the universal term with the implementation detail of the SH4 docs, where that is not the case.